src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -252,6 +270,35 @@ public static void AddPropertyIfExists(IDictionary<string, string> properties, s
 
                         restoreSummaries.AddRange(await RestoreRunner.RunAsync(restoreContext, cancellationToken));
                     }
+
+                    if (cleanupAssetsForUnsupportedProjects)
+                    {
+                        // Restore assets are normally left on disk between restores for all projects.  This can cause a condition where a project that supports PackageReference was restored
+                        // but then a user changes a branch or some other condition and now the project does not use PackageReference. Since the restore assets are left on disk, the build
+                        // consumes them which can cause build errors. The code below cleans up all of the files that we write so that they are not used during build
+                        Parallel.ForEach(dependencyGraphSpec.Projects.Where(i => !DoesProjectSupportRestore(i)), project =>
+                        {
+                            if (project.RestoreMetadata == null || string.IsNullOrWhiteSpace(project.RestoreMetadata.OutputPath) || string.IsNullOrWhiteSpace(project.RestoreMetadata.ProjectPath))
```

Comment:
I just now noticed that OutputPath is set in packagesconfig projects in static graph restore but not in the regular msbuild graph generation. 
Ofc that won't break anything cause there's no caching, but it will be something we hit if we try porting it to the current msbuild restore.